protostar/testing/pure_cairo_test_runner.py

```diff
@@ -0,0 +1,193 @@
+import asyncio
+import traceback
+from logging import getLogger
+from pathlib import Path
+from typing import List, Optional
+
+from starkware.cairo.lang.compiler.program import Program
+
+from protostar.compiler import (
+    ProjectCompiler,
+    ProjectCompilerConfig,
+)
+from protostar.compiler.project_cairo_path_builder import ProjectCairoPathBuilder
+from protostar.starknet.compiler.cairo_compilation import CairoCompiler
+from protostar.configuration_file.configuration_file_factory import (
+    ConfigurationFileFactory,
+)
+from protostar.protostar_exception import ProtostarException
+from protostar.starknet.compiler.starknet_compilation import (
+    CompilerConfig,
+)
+from . import TestRunner
+from .starkware.pure_cairo_test_execution_state import PureCairoTestExecutionState
+from .test_case_runners.pure_cairo_test_case_runner import PureCairoTestCaseRunner
+
+from .test_config import TestConfig
+from .test_environment_exceptions import ReportedException
+from .test_results import (
+    BrokenTestSuiteResult,
+    TestResult,
+    UnexpectedBrokenTestSuiteResult,
+)
+from .test_shared_tests_state import SharedTestsState
+from .test_suite import TestCase, TestSuite
+from .testing_seed import Seed
+
+
+logger = getLogger()
+
+# pylint: disable=too-many-instance-attributes
+class PureCairoTestRunner:
+    def __init__(
```

Comment:
I would name this class `CairoTestRunner` to be consistent with `CairoCompiler`

Reply:
done